File: Trabalho3/Python/lista.py
class MinhaLista:
    def __init__(self):
        self.lista = []
    
    def Insere(self, numero):
        self.lista.append(numero)
    
    def Retirar(self):
        if self.Tamanho() > 0:
            self.lista.pop()
        else:
            print("Lista esta vazia")

    def Primeiro(self):
        if self.Tamanho() > 0:
            return self.lista[0]
        else:
            print("Lista esta vazia")

    def Last(self):
        if self.Tamanho() > 0:
            return self.lista[-1]
        else:
            print("Lista esta vazia")
            return -1
        
    def Tamanho(self):
        return len(self.lista)
    
    def getLista(self):
        return self.lista
    
    def __iter__(self):
        return iter(self.lista)
    

File: Trabalho3/Java/Lista.java

package com.mycompany.listamain;

import java.util.ArrayList;
import java.util.List;


public class Lista {
    private ArrayList<Integer> lista;
    
    public Lista(){
        lista = new ArrayList<>();
    }
    
    public void insere(int numero){
        this.lista.add(numero);
    }
    
    public void retirar(){
        if(tamanho() > 0){
            lista.remove(tamanho() - 1);
        }else{
            System.out.println("ERRO, Lista vazia");
        }
    }
    
    public int primeiro(){
        if(tamanho() > 0){
            return lista.get(0);
        }else{
            System.out.println("ERRO, Lista vazia");
            return -1;
        }
    }
    
    public int last(){
        if(tamanho() > 0){
            return lista.get(tamanho());
        }else{
            System.out.println("ERRO, Lista vazia");
            return -1;
        }
    }
    
    public int tamanho(){
        return lista.size();
    }

    public List<Integer> getLista() {
        return lista;
    }
    
   
}
